### ISSUE TITLE
sshd filter does not detect "no matching mac found" messages.

### ISSUE THREAD

**Issue by BlackDex** [fail2ban/fail2ban#1943]
### Environment:

_Fill out and check (`[x]`) the boxes which apply. If your Fail2Ban version is outdated, 
and you can't verify that the issue persists in the recent release, better seek support 
from the distribution you obtained Fail2Ban from_

- Fail2Ban version (including any possible distribution suffixes): 0.10.1
- OS, including release name/version: Ubuntu 14.04.5 LTS (Trusty)
- [ ] Fail2Ban installed via OS/distribution mechanisms
- [x] Fail2Ban installed via github release package
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:

I have configure my server to only allow some specific MAC (Message authentication codes) for SSH using the "MACs" option in the sshd_config. This results in some login attempts to produce an error that there is no matching mac found for the connection. These connections are hack attempts and should be able to be blocked. Which currently it does not.

Below are the lines which shows the error message (it is on two lines with the second one has the error, and the first one has the IP of the attacker).

If you need more information please let me know.

#### Steps to reproduce
Change your MACs settings like something describe on https://github.com/stribika/stribika.github.io/blob/master/_posts/2015-01-04-secure-secure-shell.md

 Try to login with the client using other MACs set then the server.

#### Expected behavior
After the set retries and findtime it should block this attacker.

#### Observed behavior
These lines are not found.

#### Any additional information
n/a

### Configuration, dump and another helpful excerpts
n/a

#### Any customizations done to /etc/fail2ban/ configuration
n/a

#### Relevant parts of /var/log/fail2ban.log file:
n/a

#### Relevant lines from monitored log files in question:
```
Oct 25 19:37:08 hostname sshd[22477]: Connection from 1.1.1.1 port 31309 on 8.8.8.8 port 22
Oct 25 19:37:08 hostname sshd[22477]: fatal: no matching mac found: client hmac-xxx,hmac-xxx,hmac-xxx,hmac-xxx,hmac-xxx,hmac-xxx server hmac-xxx,hmac-xxx,umac-xxx,hmac-xxx,hmac-xxx,umac-xxx [preauth]
Oct 25 19:37:36 hostname sshd[22481]: Connection from 1.1.1.1 port 17707 on 8.8.8.8 port 22
Oct 25 19:37:38 hostname sshd[22481]: fatal: no matching mac found: client hmac-xxx,hmac-xxx,hmac-xxx,hmac-xxx,hmac-xxx,hmac-xxx server hmac-xxx,hmac-xxx,umac-xxx,hmac-xxx,hmac-xxx,umac-xxx [preauth]
```

**Comment by Fuchus** [comment]
The problem in this case is that your OpenSSH server is not up to scratch - version >=7.0 has a different format of logging missing key algorithms to auth.log, which includes the <HOST> in one line. That in turn can be retrieved by a correct regex formulation.
There is no easy way to fix this - just update your OpenSSH server.

**Comment by sebres** [comment]
> which includes the in one line...
> There is no easy way to fix this - just update your OpenSSH server.

This is not correct. Fail2ban could well find the multi-line failures (where the IP is not directly logged in the failure-line)...
See for example the how-to in #1966 for similar (multi-line) handling.

**Comment by sebres** [comment]
Fixed in 159957a for v.0.10 and higher. 
For multiline (buffered) variant usable in v.0.9, take a look the changes in example filter [zzz-sshd-obsolete-multiline.conf](https://github.com/fail2ban/fail2ban/commit/159957ab883efc151a2dddf53197153ba41576c2#diff-26ca86257581861d63befe5d65835816).
